Szafirmag/Zadanie5/Zadanie5b.cxx

```diff
@@ -0,0 +1,55 @@
+#include <chrono>
+#include <iostream>
+#include <thread>
+
+struct foo {
+    int x;
+    int y; 
+};
+
+static struct foo f;
+
+/* Here we have fixed problem of false
+   sharing by having functions creating
+   local copies of variables they are
+   working with.
+*/
+
+int sum_a()
+{
+    int s = 0;
+    int local_x = f.x;
+    for (int i = 0; i < 1000000; ++i)
+        s += local_x;
+    return s;
+}
+
+void inc_b()
+{
+    int local_y = f.y;
+    for (int i = 0; i < 1000000; ++i)
+        ++local_y;
+    f.y = local_y;
```

Comment:
W takim przypadku zadziała, ale gdybyśmy chcieli mimo wszystko na bieżąco aktualizować wartość f.y po każdym obrocie pętli, to należałoby użyć alignas(64). Zazwyczaj linia cache'u ma długość 64 bajtów. Dzięki temu zaobserwujesz przyspieszenie programu bez modyfikacji funkcji. Jedynie do typu int w strukturze musisz dopisać ten modyfikator. Spróbujesz i podeślesz wyniki?